core/src/main/java/io/questdb/std/str/Utf8s.java

```diff
@@ -1031,13 +1035,13 @@ public static int validateUtf8(@NotNull Utf8Sequence seq) {
     }
 
     /**
-     * Validates bytes between lo,hi addresses.
+     * Validates if the bytes between lo,hi addresses belong to a valid UTF8 sequence.
      *
-     * @return string length if input is proper UTF-8 and -1 otherwise.
+     * @return -1 if bytes are not a UTF8 sequence, 0 if this is ASCII sequence and 1 if it is non-ascii UTF8 sequence.
      */
-    public static int validateUtf8(long lo, long hi) {
-        int len = 0;
+    public static int getUtf8SequenceType(long lo, long hi) {
```

Comment:
This call is slower than a SWAR check. The purpose of the original `validateUtf8` method was to validate the input. For situations when we don't need to validate, we could introduce another, faster method with SWAR check. WDYT?

Reply:
thanks @puzpuzpuz for the review, yep, we can use another validaiton method. Do you have any suggestions?

Reply:
It depends on whether we want to validate the input or not. If we only want to check if it has non-ascii chars, we can do that faster. If we want to keep the UTF-8 validation for varchar bind variables, the current version is fine.

Let me mark this one as resolved since bind variable validation is very unlikely to be the bottleneck anyway.